fixed hardware detection

Takalele opened the pull request:
fixes https://github.com/rare-freertr/freeRtr-containerlab/issues/12

Takalele committed:
Update hwdet-init.sh

fixes hardware detection

frederic-loui commented:
Tested both dataplane `pcapInt`,  `p4emu`.

And LGTM ! 

That would be also great, if you could:

1- modify [rtr000](https://github.com/rare-freertr/freeRtr-containerlab/blob/main/rtr000.clab.yml) example  using `env`, `DATAPLANE_TYPE` in containerlab topology file as you did on README.md and add this change to the PR. 

1- ADD a default VRF for sdn interface binding. (We can call it CORE or another name of your choice)
``` 
conf t
!
vrf definition CORE
  exit
!
interface sdn1
  vrf forwarding CORE
  exit
!
server p4lang p4
  export-vrf CORE
!
```
The idea is to underline the fact that `p4emu` dataplane export physical interface as `sdn` port that belongs to a `VRF` or `BRIDGE` . You can obviously keep this VRF definition for `pcapInt` dataplane flavor.

In case people "miss" to read the README.md 😄 

Thanks !

Takalele committed:
Update README.md

Takalele committed:
Update hwdet-init.sh

default vrf initialization

Takalele committed:
Update rtr000.clab.yml

VRF Initialization

Takalele commented:
I have implemented the requested changes:

1. Modified the rtr000 example to use the DATAPLANE_TYPE and the new INIT_VRF environment variable in the containerlab topology file.

2. Implemented automatic VRF initialization (CORE) which:
   - Creates the VRF definition
   - Adds VRF forwarding to ALL interfaces (except ethernet0 and ethernet255)
   - For P4 dataplane: adds export-vrf CORE to the server p4lang p4 section and exports bridges

3. Updated the README.md to document the new INIT_VRF functionality, including configuration options and an example.

BR
Andreas

frederic-loui commented:
Hi !

All is working as you described.
Thanks for this PR update !

tiny detail before merging:
When `DATAPLANE_TYPE=pcapInt` pcapInt and `INIT_VRF="true"` true then `ethernet1` should be bound to `CORE` VRF defined previously. 
(This is optional but shows that freertr VRF/BRIDGE concepts are strongly rooted into freertr cli configuration philosophy)

All the best 
Frederic

Takalele committed:
- updated the scripts to use the freertr buildin hardware detection and externalisation

Takalele committed:
- updated the scripts to use the freertr buildin hardware detection and externalisation

Takalele committed:
- updated the scripts to use the freertr buildin hardware detection and externalisation

Takalele commented:
hi @frederic-loui,

now the hardware detection is using the freertr built-in knobs as @mc36 recommended. it is now possible to start with pcapint and change it to something else like p4emu without rebooting the container. however, the dataplane will be reset to pcapint after redeployment, as long as you saved the configuration it should persist. reverting back to pcapint is only possible by redeploying the lab.

all other dataplanes are now compatible as long as they work with docker, such as p4udp.

the init_vrf vrf bug you mentioned has also been fixed.

additionally, mldsakey has been added to the default.cfg.

```
write
test hwext path /rtr/run/conf/rtr- dataplane p4emu
reload force
```


```
ssh rare@clab-rtr000-rtr2
Warning: Permanently added 'clab-rtr000-rtr2' (RSA) to the list of known hosts.
rare@clab-rtr000-rtr2's password:
welcome
line ready
rtr2#show inter status
interface    state  mtu   promisc  macsec  sgt    changes  for       since
ethernet1    up     1500  false    false   false  3        00:00:12  2025-07-04 22:24:32
ethernet2    up     1500  false    false   false  3        00:00:12  2025-07-04 22:24:32
ethernet255  up     1500  false    false   false  3        00:00:12  2025-07-04 22:24:32

rtr2#conf t
rtr2(cfg)#int ethernet1
rtr2(cfg-if)#cdp enable
rtr2(cfg-if)#exit
rtr2(cfg)#int eth2
rtr2(cfg-if)#cdp enable
rtr2(cfg-if)#end
rtr2#wr
% saving configuration
% success
rtr2#test hwext path /rtr/run/conf/rtr- dataplane p4emu
% externalizing forwarding
% found 3 interfaces
% ifc 0: sdn1 ethernet1 eth1 aa:c1:ab:c9:c1:fd
% ifc 1: sdn2 ethernet2 eth2 aa:c1:ab:1c:23:e1
% ifc 2: sdn3 ethernet255 eth0 42:54:aa:b1:d4:f0
% found 2 vrfs and 0 bridges
% vrf 0: CORE
% vrf 1: OOB
% finished
rtr2#reload force
Connection to clab-rtr000-rtr2 closed by remote host.
Connection to clab-rtr000-rtr2 closed.

ssh rare@clab-rtr000-rtr2
Warning: Permanently added 'clab-rtr000-rtr2' (RSA) to the list of known hosts.
rare@clab-rtr000-rtr2's password:
welcome
line ready
rtr2#show runn int sdn1
interface sdn1
 mtu 1500
 macaddr aac1.abc9.c1fd
 cdp enable
 vrf forwarding CORE
 no shutdown
 no log-link-change
 exit
!
rtr2#show runn int sdn2
interface sdn2
 mtu 1500
 macaddr aac1.ab1c.23e1
 cdp enable
 vrf forwarding CORE
 no shutdown
 no log-link-change
 exit
!
rtr2#show running-config server p4lang p4
server p4lang p4
 export-vrf CORE
 export-vrf OOB
 export-port sdn1 0 0 0 0 0
 export-port sdn2 1 0 0 0 0
 export-port sdn255 2 0 0 0 0
 interconnect ethernet0
 vrf p4
 exit
!
```

BR
Andreas

Takalele committed:
removed function rename_mgmt_interface - there is no need for the function anymore
